damnit/gui/widgets.py

```diff
@@ -260,3 +263,170 @@ def currentLineColor(self, countDistance, totalNrOfLines, trailFadePerc, minOpac
             resultAlpha = min(1.0, max(0.0, resultAlpha))
             color.setAlphaF(resultAlpha)
         return color
+
+
+class PlotLineWidget(QWidget):
+    def __init__(self, x_data, y_data):
+        super().__init__()
+        self.x_data = x_data
+        self.y_data = y_data
+        self.slider_position = (x_data[0], x_data[-1])  # (left, right)
+        self.setMinimumSize(100, 100)
+
+    def set_slider_position(self, position: tuple[float, float]):
+        self.slider_position = position
+        self.update()
+
+    def paintEvent(self, event):
```

Comment:
I see calls to `.repaint()` in the test code, but it looks like this method isn't getting executed in the tests. Is there a good way to remedy that?